src/DynamoCore/Nodes/VariableInputNode.cs

```diff
@@ -73,13 +73,13 @@ public override int GetInputIndexFromModel()
         protected virtual void RemoveInput()
         {
             VariableInputController.RemoveInputBase();
-            OnAstUpdated();
+            //OnAstUpdated();
```

Comment:
Are we sure we want these `OnAstUpdated` removed? For example, for `List.Create`, removing one input port while it connects to something will remove the same element from the result. Please try that out to confirm this continues to work.